tests/baselines/reference/keyofAndForIn.errors.txt

```diff
@@ -0,0 +1,44 @@
+keyofAndForIn.ts(4,10): error TS2405: The left-hand side of a 'for...in' statement must be of type 'string' or 'any'.
+keyofAndForIn.ts(15,10): error TS2405: The left-hand side of a 'for...in' statement must be of type 'string' or 'any'.
+keyofAndForIn.ts(26,10): error TS2405: The left-hand side of a 'for...in' statement must be of type 'string' or 'any'.
+
+
+==== keyofAndForIn.ts (3 errors) ====
+    function f1<K extends string, T>(obj: { [P in K]: T }, k: K) {
+        const b = k in obj;
+        let k1: K;
+        for (k1 in obj) {
+             ~~
+!!! error TS2405: The left-hand side of a 'for...in' statement must be of type 'string' or 'any'.

```

Comment:
This should be an error because this is a valid call:
```ts
declare const obj: { a: 1, b: 2, c: 3 }
declare const key: 'a' | 'b'
f1(obj, key)
```

`'c'` is not assignable to `K`